code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm

```diff
@@ -122,9 +122,7 @@
 
 	user.visible_message("<span class='warning'>[user] pins [M] into [src], preparing the securing resin.</span>",
 	"<span class='warning'>[user] pins [M] into [src], preparing the securing resin.</span>")
-	var/M_loc = M.loc
-	if(do_after(user, 15, TRUE, 5, BUSY_ICON_HOSTILE))
-		if(M.loc != M_loc) return
+	if(do_after(user, 15, TRUE, M) && user.Adjacent(src))
```

Comment:
Is the `Adjacent` check necesssary, if the `do_after` proc fails if the target's loc changes to something other than the `user` or the `user.loc`?